### DISCUSSION TITLE
Model observers

### DISCUSSION THREAD

**Discussion by AvantaR** [prisma/prisma#9890] [Q&A] [answered]
Hi!

Does Prisma dispatch any events in model's lifecycle? I'm looking for something similar to [Laravel Eloquent Observers](https://laravel.com/docs/8.x/eloquent#observers). I would like to hook up to e.g. created event on User model, and then, when a new User is created, run some code. I know that I can create a middleware, but I'm not sure it's the best way deal with it.

Thanks in advance for your help!

**Comment by austincrim** [comment, marked as answer]
Hi @AvantaR! 

Currently, [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) would be the way to accomplish what you want. If you are interested in a more robust event system in Prisma, please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Reply by AvantaR** [reply]
Thanks for your help :smile:

**Comment by AvantaR** [comment]
I've found some time and prepare a small middleware package, which more or less provides similar feature for Prisma. Maybe it will be useful for someone else as well.

https://www.npmjs.com/package/prisma-event-dispatcher

Cheers!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
